components/service-mgt/axis2-service-mgt/org.wso2.carbon.service.mgt/src/main/java/org/wso2/carbon/service/mgt/ServiceAdmin.java

```diff
@@ -2544,22 +2546,26 @@ private void sendClusterSyncMessage(ServiceConstants.ServiceOperationType applic
 
     private boolean isAxisServiceCApp(AxisService axisService) {
         //Check if Service is deployed from a CApp
-        if (AXIS2_SERVICE_TYPE.equals(getServiceType(axisService))) {
-            try {
-                Path axis2ServiceAppPath = Paths.get(axisService.getFileName().toURI());
-                if (axis2ServiceAppPath != null) {
-                    String tenantId = AppDeployerUtils.getTenantIdString();
-                    // Check whether there is an application in the system from the given name
-                    ArrayList<CarbonApplication> appList = DataHolder.getApplicationManager().getCarbonApps(tenantId);
-                    for (CarbonApplication application : appList) {
-                        Path cappPath = Paths.get(application.getExtractedPath());
-                        if (axis2ServiceAppPath.startsWith(cappPath)) {
-                            return true;
+        if (AXIS2_SERVICE_TYPE.equals(getServiceType(axisService)) || PROXY_SERVICE_TYPE
```

Comment:
Shall we check whether we need this condition

Reply:
What will happen to APIs ?

Reply:
For artifacts other than proxies that are deployed via capp, the delete function is disabled.